core/conversion/converters/converter_util.cpp

```diff
@@ -0,0 +1,52 @@
+#include "core/conversion/converters/converter_util.h"
+#include "core/conversion/converters/converters.h"
+#include "core/util/prelude.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+
+nvinfer1::ILayer* addPaddingLayer(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor* tensor, int nDim, bool trailing, bool use_zeros) {
+  const auto dims = tensor->getDimensions();
+
+  if (dims.nbDims < nDim) {
+    auto newDims = dims;
+    for (int dim = dims.nbDims; dim < nDim; ++dim)
+      newDims = util::unsqueezeDims(newDims, trailing ? dim : 0, 1, use_zeros);
+    LOG_DEBUG(
+        "Input shape is less than 4D got: " << dims
+                                            << ", inserting shuffle layer to reshape to 4D tensor shape: " << newDims);
+
+    LOG_DEBUG("Original shape: " << dims << ", reshaping to: " << newDims);
+    auto shuffle_layer = ctx->net->addShuffle(*tensor);
+    TRTORCH_CHECK(shuffle_layer, "Unable to create shuffle layer");
+    shuffle_layer->setReshapeDimensions(newDims);
+    shuffle_layer->setZeroIsPlaceholder(use_zeros);
+    shuffle_layer->setName((util::node_info(n) + " [Reshape to " + util::toStr(newDims) + ']').c_str());
+    return shuffle_layer;
+  } else
+    return nullptr;
```

Comment:
Not sure about returning `nullptr`. Could we maybe return a ITensor* instead of an ILayer* and pass through the original tensor if padding is not needed? Or we could maybe just assert `dims.nbDims < nDim` and fail?

Reply:
What's wrong with returning nullptr ? The idea is that you call it on 3 or 4 dimensional input and it returns shuffle layer if shuffle is needed and nullptr otherwise. If we operate on Tensors and/or consider 4d input to be error, that would put conditions in the caller (which may be a good thing making it more explicit ), but would hide shuffle layer from the caller. Also, net->addXxx methods return ILayers, so I thought it would be more intuitive.

Reply:
Really its about reducing sharp edges. In theory power users of TRTorch might interact with these tools in either their own converters or as contributors. Conditionally returning `nullptr` could cause components missing the logic to handle it or with poor testing coverage to fail unexpectedly (to the user) and with a segfault vs some descriptive error. 

"that would put conditions in the caller (which may be a good thing making it more explicit )". Yeah agree, id rather add caller conditions and be explicit. So its on the converter writer to decided if a shuffle layer is needed. Or we can set clear conditions for usage and if those conditions aren't met we error out with a meaningful message so its clear what happened. 

RE: ILayer v. ITensor, I think the argument for tensor is that I dont think people will do a whole lot extra with the ILayer itself (name is set for them and all the settings are set for them). So really we are just asking people to add the `layer->getOutput(0)` boiler plate. I think theres a couple other places (tensor_to_const) where we dont bother returning the layer we just return the relevant ITensor since thats the piece that people really care about.

Reply:
I actually deleted my initial comment - I guess, not fast enough :) Anyway, it does look better now that I followed your suggestion. I have removed 'Layer' from the name, too.